server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java

```diff
@@ -165,7 +166,10 @@ private void checkAdminPermission(Optional<ComponentDto> component) {
 
   private Map<String, String> getKeysToDisplayMap(Set<String> keys) {
     return keys.stream()
-      .collect(Collectors.toMap(propertyDefinitions::validKey, Function.identity()));
+      .collect(Collectors.toMap(propertyDefinitions::validKey, Function.identity(),
+        (u, v) -> {
+          throw new IllegalArgumentException(format("'%s' and '%s' cannot be used at the same time as they refer to the same setting", u, v));
```

Comment:
In fact I just override what is already done in the default merge function, I just replace the IllegalStateException by an IllegalArgumentException and set a more suitable message.